include/SFML/Window/Cursor.hpp

```diff
@@ -0,0 +1,219 @@
+    Cursor();
+
+    ////////////////////////////////////////////////////////////
+    /// \brief Destructor
+    ///
+    /// This destructor releases the system resources
+    /// associated with this cursor, if any.
+    ///
+    ////////////////////////////////////////////////////////////
+    ~Cursor();
+
+    ////////////////////////////////////////////////////////////
+    /// \brief Create a cursor with the provided image
+    ///
+    /// \a pixels must be an array of \a width by \a height pixels
+    /// in 32-bit RGBA format. If not, this will cause undefined behavior.
+    ///
+    /// If \a pixels is null or either \a width or \a height are 0,
+    /// the current cursor is left unchanged and the function will
+    /// return false.
+    ///
+    /// In addition to specifying the pixel data, you can also
+    /// specify the location of the hotspot of the cursor. The
+    /// hotspot is the pixel coordinate within the cursor image
+    /// which will be located exactly where the mouse pointer
+    /// position is. Any mouse actions that are performed will
+    /// return the window/screen location of the hotspot.
+    ///
+    /// \warning On Unix, the pixels are mapped into a monochrome
+    ///          bitmap: pixels with an alpha channel to 0 are
+    ///          transparent, black if the RGB channel are close
+    ///          to zero, and white otherwise.
+    ///
+    /// \param pixels   Array of pixels of the image
+    /// \param size     Width and height of the image
+    /// \param hotspot  (x,y) location of the hotspot
+    /// \return true if the cursor was successfully loaded;
+    ///         false otherwise
+    ///
+    ////////////////////////////////////////////////////////////
+    bool loadFromPixels(const Uint8* pixels, Vector2u size, Vector2u hotspot);
+
+    ////////////////////////////////////////////////////////////
+    /// \brief Create a native system cursor
+    ///
+    /// Refer to the list of cursor available on each system
+    /// (see sf::Cursor::Type) to know whether a given cursor is
+    /// expected to load successfully or is not supported by
+    /// the operating system.
+    ///
+    /// \param type Native system cursor type
+    /// \return true if and only if the corresponding cursor is
+    ///         natively supported by the operating system;
+    ///         false otherwise
+    ///
+    ////////////////////////////////////////////////////////////
+    bool loadFromSystem(Type type);
+
+private:
+
+    friend class Window;
+
+    ////////////////////////////////////////////////////////////
+    /// \brief Get access to the underlying implementation
+    ///
+    /// This is primarily designed for sf::Window::setMouseCursor,
+    /// hence the friendship.
+    ///
+    /// \return a reference to the OS-specific implementation
+    ///
+    ////////////////////////////////////////////////////////////
+    const priv::CursorImpl& getImpl() const;
```

Comment:
Ah, that's just to dispatch in one place the actual `CursorImpl` for the `WindowImpl`. I think we didn't need this before because all platform dependent types were already "inside" `WindowImpl`, but this is not the case here.